src/controller/CHIPDeviceController.cpp

```diff
@@ -1020,6 +1023,114 @@ CHIP_ERROR DeviceCommissioner::UnpairDevice(NodeId remoteDeviceId)
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR DeviceCommissioner::SendEchoRequest(NodeId remoteDeviceId)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    Messaging::SendFlags sendFlags;
+    System::PacketBufferHandle payloadBuf;
+    char * requestData = nullptr;
+
+    uint32_t size = mPingParams.GetEchoReqSize();
+    VerifyOrExit(size <= kMaxPayloadSize, err = CHIP_ERROR_MESSAGE_TOO_LONG);
+
+    requestData = static_cast<char *>(chip::Platform::MemoryAlloc(size));
+    VerifyOrExit(requestData != nullptr, err = CHIP_ERROR_NO_MEMORY);
+
+    snprintf(requestData, size, "Echo Message %" PRIu32 "\n", mPingParams.GetEchoReqCount() + 1);
```

Comment:
Why not use the payload buffer in-place below?

Reply:
acked